java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -3375,14 +3389,17 @@ static ColumnVector createColumnVector(DType type, int rows,
         mainValidDevBuff = DeviceMemoryBuffer.allocate(validLen);
         mainValidDevBuff.copyFromHostBuffer(mainColValid, 0, validLen);
       }
-      // data is null
+      if (data != null) {
+        long dataLen = data.length;
+        mainDataDevBuff = DeviceMemoryBuffer.allocate(dataLen);
+        mainDataDevBuff.copyFromHostBuffer(data, 0, dataLen);
+      }
       if (mainColOffsets != null) {
-        long offsetsLen = OFFSET_SIZE * (mainColRows + 1);
```

Comment:
Can we please revert this change?  When copying data from the host to the device this was here to truncate the buffer, if needed, and it is now lost.

Also while you are at it could we add in a comment about why we are doing it this way.

Reply:
My apologies. It works without this change of course and as you said, is needed. I have a couple things from my testing that I need to do a pass through over and this is certainly one of those that I missed. Thanks for catching that.